core/src/channel/channel_sf.rs

```diff
@@ -28,58 +29,73 @@ impl ChannelSoundfont {
         ChannelSoundfont {
             soundfonts: Vec::new(),
             matrix: VoiceSpawnerMatrix::new(),
-            curr_bank: None,
-            curr_preset: None,
+            curr_bank: 0,
+            curr_preset: 0,
         }
     }
 
     pub fn set_soundfonts(&mut self, soundfonts: Vec<Arc<dyn SoundfontBase>>) {
-        self.soundfonts = soundfonts;
-        self.curr_bank = None;
-        self.curr_preset = None;
-        self.rebuild_matrix(0, 0);
+        if !are_arc_vecs_equal(&self.soundfonts, &soundfonts) {
+            self.soundfonts = soundfonts;
```

Comment:
Nitpick but `rebuild_matrix` sets self.bank and self.preset, but self.soundfonts is set by this function. Would be nice to be more consistent with this.

Reply:
Changed this, let me know if this is what you meant.

Will do some more testing and a few benchmarks and if it's alright I'll merge.

Reply:
Yeah that works